app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -310,7 +311,11 @@ internal class MessagesActivity :
         nManager.cancelAll()
         val filter = IntentFilter()
         filter.addAction(WebSocketService.NEW_MESSAGE_BROADCAST)
-        registerReceiver(receiver, filter)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            registerReceiver(receiver, filter, RECEIVER_NOT_EXPORTED)
```

Comment:
When using an sdk34 emulator, the messages aren't updated automatically. When using RECEIVER_EXPORTED here it seems to work. Can you reproduce this?

[message.webm](https://github.com/gotify/android/assets/14895212/8ad7133c-4d7b-40aa-808c-deebc6919cf0)

Reply:
You are right. That's most likely due to changes in the intent system in Android 14.  
Exporting the receiver is not needed in our case as we are only transfering data inside Gotify.  
That's btw most probably the reason for duplicate entries in the debug and release apps, the global broadcast has always been received by both app versions and interpreted twice.

But I could not get it working yet without exposing the receiver. Then I found a very good alternative, the local broadcasting systems.  
It offers the advantage of not exposing any data to every other app as it's only processed inside Gotify. Furthermore we prevent duplicate entries as mentioned in https://github.com/gotify/android/issues/242 and it improves performance.

Are there reasons why you chose global broadcasting back then or is the local broadcasting enough for this app?  
If it's enough, the problem is solved :)

Reply:
It works again with the implementation in https://github.com/gotify/android/pull/325/commits/f4279389c62fe81f59c75a78d3d1a4a4858234e8

Reply:
> Are there reasons why you chose global broadcasting back then

Back then no, but its more or less evolved into a feature with the mention in here https://github.com/gotify/android/issues/44#issuecomment-808758340

If possible I'd like to not break this.

Reply:
Hmm, I see. Then we have no chance but sending the broadcast to all other apps (and exposing all incoming messages).  
I've switched it to RECEIVER_EXPORTED.

As an addition I would recommend to send targeted intent broadcasts or a "debug" intent broadcast (just a different action name) in debug configuration so the messages of the debug app version only appear there and don't lead to duplicate messages, what do you think?

Reply:
Yeah, sounds good!